### ISSUE TITLE
Compile error for generic type with nested immutable object. (<any> generated) 

### ISSUE THREAD

**Issue by tolgaakkoca** [immutables/immutables#1490]
Hello,
I've seen this issue here #976 and related ones. Comments looks like it is resolved with new releases. I am facing the same error in the relatively new version of immutables (2.9.3). Here is the details:

- I am using immutables library version 2.9.3. I want to upgrade the Java version of my application from 11 to 17. 
- I am using gradle 8.2.1
- Error does not occur in Java 11. As I change the JDK to 17, compile error occurs. 

1. First class:
```
@Value.Immutable
public abstract class FirstObject<B extends Serializable> {
    public abstract Optional<B> getItem();
}
```
3. Second class:
```
@Value.Immutable
public abstract class SecondObject<T extends Serializable> {
    public abstract ImmutableFirstObject<T> getFirstObject(); // ImmutableFirstObject is generated immutable object from FirstObject
}
```
5. Generated Immutable object from second object:
```
/**
 * Immutable implementation of {@link SecondObject}.
 * <p>
 * Use the builder to create immutable instances:
 * {@code ImmutableSecondObject.builder()}.
 */
@Generated(from = "SecondObject", generator = "Immutables")
@SuppressWarnings({"all"})
@javax.annotation.processing.Generated("org.immutables.processor.ProxyProcessor")
public final class ImmutableSecondObject<T extends Serializable>
    extends SecondObject<T> {
  private final <any> firstObject;

  private ImmutableSecondObject(<any> firstObject) {
    this.firstObject = firstObject;
  }
```

At compile time there occurs an error for `<any>` generic type which is generated wrong. I wonder if I can handle this problem with any solution. Thanks in advance for your support.

**Comment by elucash** [comment]
(duplicate  https://github.com/immutables/immutables/issues/1380)
It is a problem with not-yet-generated type `ImmutableFirstObject`. I would suggest, at some point to switch to using only abstract value type in signatures and only use references to generated types in code (inside methods) to instantiate / build instances. All that worked relatively well in Java6–8 and Maven, after that compilers changed a lot and also build tools (like Gradle) introduce more sophisticated ways to isolate annotation processors so this (specifically: using not-yet-generated type in signatures, as attribute types of other values, as params or return types etc) is no longer works reliably in the face of Java / tools upgrades. Need to update documentation to reflect this, but anyway. Currently this is a territory of undefined behavior on part of compiler: should it retain information about not found symbol and keep it as was specified in the source code (just having and error flag/type) or turn it into obscure `<any>`

I would suggest to try newly released `2.10.0` it has some fixes, possibly related. Also, for build tools it helps to try

1. Switch off incremental compilation or any cool isolation mode for the annotation processor
2. In Gradle, use `options.sourcepath` on Javac configuration to point to the sources root, this might help Immutables to peek into sources (during compilation) and see un-mangled type (instead of `<any>`)

**Comment by tolgaakkoca** [comment]
Hi @elucash , thanks for suggestion. The problem has been resolved when using `options.sourcepath` to point to the sources root.

`options.sourcepath = files(sourceSets.main.java.srcDirs)`
